src/libraries/Microsoft.PowerFx.Core/Syntax/Visitors/DTypeVisitor.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.PowerFx.Core.Binding.BindInfo;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Syntax.Visitors
+{
+    internal class DTypeVisitor : TexlFunctionalVisitor<DType, DefinedTypeSymbolTable>
+    {
+        public override DType Visit(ErrorNode node, DefinedTypeSymbolTable context)
+        {
+            throw new NotImplementedException();
```

Comment:
We may find partial results useful. Might be worth having this take an Error container, add an error and return `DType.Error` for all these notimpls?
Maybe not in this PR, but something to consider.

Reply:
I.e, maybe getting a type out of `Type({Foo: Number, Bar: <Err>})` while also adding an error to the expression might be useful.